src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py

```diff
@@ -1,71 +1,78 @@
+from abdeladim_2023nwbconverter import Abdeladim2023NWBConverter
+from abdeladim_2023nwbconverter import get_default_segmentation_to_imaging_name_mapping
 
+def session_to_nwb(
+    data_dir_path: Union[str, Path], output_dir_path: Union[str, Path], session_id: str, stub_test: bool = False
+):
     data_dir_path = Path(data_dir_path)
     output_dir_path = Path(output_dir_path)
     if stub_test:
         output_dir_path = output_dir_path / "nwb_stub"
     output_dir_path.mkdir(parents=True, exist_ok=True)
-
-    session_id = "subject_identifier_usually"
+    subject_id = "unknown"  # TODO ask for subject_id
     nwbfile_path = output_dir_path / f"{session_id}.nwb"
 
-    source_data = dict()
-    conversion_options = dict()
+    # Add Imaging
+    imaging_folder_path = data_dir_path / "raw-tiffs" / session_id
+    # Add Segmentation
+    segmentation_folder_path = data_dir_path / "processed-suite2p-data/suite2p"
 
-    # Add Recording
-    source_data.update(dict(Recording=dict()))
-    conversion_options.update(dict(Recording=dict()))
+    segmentation_to_imaging_plane_map = get_default_segmentation_to_imaging_name_mapping(imaging_folder_path, segmentation_folder_path)
 
-    # Add LFP
-    source_data.update(dict(LFP=dict()))
-    conversion_options.update(dict(LFP=dict()))
-
-    # Add Sorting
-    source_data.update(dict(Sorting=dict()))
-    conversion_options.update(dict(Sorting=dict()))
-
-    # Add Behavior
-    source_data.update(dict(Behavior=dict()))
-    conversion_options.update(dict(Behavior=dict()))
+    converter = Abdeladim2023NWBConverter(
+        imaging_folder_path=imaging_folder_path,
+        segmentation_folder_path=segmentation_folder_path,
+        segmentation_to_imaging_map=segmentation_to_imaging_plane_map,
+        verbose=False,
+    )
 
-    converter = Abdeladim2023NWBConverter(source_data=source_data)
+    conversion_options = {
+        interface_name: dict(stub_test=stub_test) for interface_name in converter.data_interface_objects.keys()
+    }
+    photon_series_index = 0
+    for interface_name in converter.data_interface_objects.keys():
+        if "Imaging" in interface_name:
+            conversion_options[interface_name]={"photon_series_index":photon_series_index}     
+            photon_series_index += 1
 
     # Add datetime to conversion
     metadata = converter.get_metadata()
-    datetime.datetime(
-        year=2020, month=1, day=1, tzinfo=ZoneInfo("US/Eastern")
-    )
-    date = datetime.datetime.today()  # TO-DO: Get this from author
-    metadata["NWBFile"]["session_start_time"] = date
 
     # Update default metadata with the editable in the corresponding yaml file
     editable_metadata_path = Path(__file__).parent / "abdeladim_2023_metadata.yaml"
     editable_metadata = load_dict_from_file(editable_metadata_path)
     metadata = dict_deep_update(metadata, editable_metadata)
-
+    metadata = dict_deep_update(metadata, {"Subject": {"subject_id": subject_id}})
```

Comment:
I think there is no need to call `dict_deep_update` to add the subject metadata, also where do you update the session_id?
```suggestion
    metadata["Subject"].update(subject_id=subject_id)
```

Reply:
Ops, I forgot to update session_id. Thanks